tests/filters.test.js

```diff
@@ -181,4 +181,94 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('makeBuildStaticAddonCards', () => {
+    const STATIC_ADDON_CARD = '<!-- static add-on card -->';
+
+    it('returns a function', () => {
+      expect(makeBuildStaticAddonCards()).toBeInstanceOf(Function);
+    });
+
+    it('handles content without an add-on card', async () => {
+      const content = 'some content';
+      const callback = jest.fn();
+      const _buildStaticAddonCard = jest.fn();
+
+      await makeBuildStaticAddonCards({ _buildStaticAddonCard })(
+        content,
+        callback
+      );
+
+      expect(callback).toHaveBeenCalledWith(null, content);
+      expect(_buildStaticAddonCard).not.toHaveBeenCalled();
+    });
+
+    it('calls the card library to generate a card', async () => {
+      const addonId = '1234';
+      const content = `<div class="addon-card" data-addon-id="${addonId}"></div>`;
+      const callback = jest.fn();
+      const _buildStaticAddonCard = jest
+        .fn()
+        .mockReturnValue(STATIC_ADDON_CARD);
+
+      await makeBuildStaticAddonCards({ _buildStaticAddonCard })(
+        content,
+        callback
+      );
+
+      expect(callback).toHaveBeenCalledWith(null, STATIC_ADDON_CARD);
+      expect(_buildStaticAddonCard).toHaveBeenCalledWith({ addonId });
+    });
+
+    it('can replace multiple add-on cards', async () => {
+      const addonIds = ['12', '34', '56'];
+      const content = addonIds
+        .map(
+          (addonId) =>
+            `<div class="addon-card" data-addon-id="${addonId}"></div>`
+        )
+        .join('\n');
+      const callback = jest.fn();
+      const _buildStaticAddonCard = jest
+        .fn()
+        .mockReturnValue(STATIC_ADDON_CARD);
+
+      await makeBuildStaticAddonCards({ _buildStaticAddonCard })(
+        content,
+        callback
+      );
+
+      expect(callback).toHaveBeenCalledWith(
+        null,
+        [STATIC_ADDON_CARD, STATIC_ADDON_CARD, STATIC_ADDON_CARD].join('\n')
+      );
+      for (const addonId of addonIds) {
+        expect(_buildStaticAddonCard).toHaveBeenCalledWith({ addonId });
+      }
+    });
+
+    it('handles errors coming from the addons-frontend-card library', async () => {
+      const addonId = '1';
+      const content = [
+        'content before',
+        `<div class="addon-card" data-addon-id="${addonId}"></div>`,
+        'content after',
+      ].join('\n');
+      const callback = jest.fn();
+      const _buildStaticAddonCard = jest.fn().mockImplementation(() => {
+        throw new Error('error coming from the addons-frontend-card library');
+      });
+
+      await makeBuildStaticAddonCards({ _buildStaticAddonCard })(
+        content,
+        callback
+      );
+
+      expect(callback).toHaveBeenCalledWith(
+        null,
+        'content before\n\ncontent after'
```

Comment:
I wonder if the lib throws should this break the build rather than replacing the card entry with nothing? Or alternatively should there be some content showing an error message. If the copy around the cards refer to it, it might be weird if there's no card.

Reply:
This could totally be checked and handled in a follow-up, it doesn't need to block landing this.

Reply:
That will depend on the mock-ups indeed. The latest ones show a card with a message saying "extension not available".

Reply:
> If the copy around the cards refer to it, it might be weird if there's no card.

This should be double-checked with Editorial/Product because I heard we could potentially hide the card fully at some point. That being said, UX created a "unavailable" version of the card so we might not need that.